CDCgov/prime-reportstream
# Adds additional metadata columns for ELR and fixes bug with test performed fields

This PR adds additional metadata columns for ELR and fixes bug with test performed fields. In specific we are looking to normalize some fields that are coming in with different values from the senders. For example, the test result text field (`OBX-5-2`) has different values per sender, even if they're using the correct code value in `OBX-5-1`. This causes some challenges and discrepancies in reading the data. To that end I've added in a column that normalizes the sender's value to the preferred values for the SNOMED code.

This has also been done for ordered test and performed test data, and I've added a column to collect the long name for the test based on the test code being used.

It also adds the specimen type code, specimen source site code, specimen collection method code, and specimen collection site code to the metadata table so we can group by those values.

It makes some minor convention changes for null-ing out empty fields, updates some of the normalized fields I've added so historical data is filled in, and corrects a bug where the test result and test result code fields were swapped.

Test Steps:
1. Build your local container, and make sure that the V55 script has run against your DB AND jooq has rebuilt. You may need to issue a `gradlew clean` if you encounter errors.
2. Run the application
3. Submit the attached test file to your local instance `curl -X POST -H 'client: mayo-clinic' -H 'Content-Type: application/hl7-v2' --data-binary @"<PATH TO FILE>" "$LOCAL_PRIME/reports"`
4. Once it has completed submission review the contents the `elr_result_metadata` table in your local DB and verify that the results you see in the table match the attached file.

## Changes
- Added new flyway script for migration
- Added new valuesets to use for normalization when saving the metadata
- Wired in the additional fields being added to the `elr_result_metadata` table
- Updated the schemas to normalize some fields (adding code vs name to some fields for clarity)
- Added `trimToNull` call on the `getStringOrNull` method to enforce nullity for empty string values in the DB
- Generated documentation

## Checklist

### Testing

### Attachments
[monkeypox-e0386067-6d1e-4d6a-b210-c9a4f5862f29-20220720190541.hl7.txt](https://github.com/CDCgov/prime-reportstream/files/9304265/monkeypox-e0386067-6d1e-4d6a-b210-c9a4f5862f29-20220720190541.hl7.txt)
